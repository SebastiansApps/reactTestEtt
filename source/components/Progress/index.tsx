import * as React from 'react';
import{View, Text } from 'react-native';

interface IProps {
    amount: number;
}

const Progress = ({amount}: IProps) => {
    return (
    <View>
        <Text>Progress: {amount}</Text>
    </View>)
}
export default Progress;