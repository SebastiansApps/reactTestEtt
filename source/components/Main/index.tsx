import React, { Component } from 'react';
import { View, Button, Text, TextInput, Alert } from 'react-native';
import Progress from '../Progress';
import { Actions } from 'react-native-router-flux';

interface IState {
    isLoaded: boolean;
    userName: string;
    amountClicks: number;
}

export default class App extends Component <{}, IState> {
// PROPS, FUNC:

constructor(props: any) {
  super(props);
  this.state = {
    isLoaded: false,
    userName: '',
    amountClicks: 0,
};
}

onTextChanged(newName: string) {
  this.setState({
    userName: newName,
    amountClicks: this.state.amountClicks + 1,
  });
}

componentDidMount() {
  Actions.push('login');
}

  render() {
    return (
      <View>
        <TextInput
        style={{padding: 40, borderColor: 'gray', borderWidth: 1, fontSize: 14}}
        placeholder='Type here pleass'
        onChangeText ={(text) => this.onTextChanged(text)}
        />
        <Text
        style={{fontSize: 14, padding: 30}}>
        {'Welcome' + ' ' + this.state.userName}</Text>
        <Button
          onPress={() => {
            Alert.alert('Error');
          }}
          title='Next'
          />

          <Progress amount={this.state.amountClicks} />

      </View>
    );
  }
}
