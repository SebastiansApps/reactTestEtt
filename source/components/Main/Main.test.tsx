/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';
import Progress from '../Progress';
// import Login from '../Login';
// import TextInput till react

import Main from './';

function setup() {
  const props = {
// Här kan ni mocka komponentens props om sådana finns
// typ userName: 'test'
  };

  const enzymeWrapper = shallow(<Main {...props} /> );

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {

  describe('Main', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();

      const progress = enzymeWrapper.find(Progress).props();
      expect(progress.amount).toBe(0);

      const login = enzymeWrapper.find(textInput).at(1).props;
      expect();
    });
  });
});