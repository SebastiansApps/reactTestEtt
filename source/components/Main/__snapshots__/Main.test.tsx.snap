// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Main should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          onChangeText={[Function]}
          placeholder="Type here pleass"
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "fontSize": 14,
              "padding": 40,
            }
          }
        />,
        <UNDEFINED
          style={
            Object {
              "fontSize": 14,
              "padding": 30,
            }
          }
        >
          Welcome 
        </UNDEFINED>,
        <UNDEFINED
          onPress={[Function]}
          title="Next"
        />,
        <Progress
          amount={0}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeText": [Function],
          "placeholder": "Type here pleass",
          "style": Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "fontSize": 14,
            "padding": 40,
          },
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Welcome ",
          "style": Object {
            "fontSize": 14,
            "padding": 30,
          },
        },
        "ref": null,
        "rendered": "Welcome ",
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "title": "Next",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            onChangeText={[Function]}
            placeholder="Type here pleass"
            style={
              Object {
                "borderColor": "gray",
                "borderWidth": 1,
                "fontSize": 14,
                "padding": 40,
              }
            }
          />,
          <UNDEFINED
            style={
              Object {
                "fontSize": 14,
                "padding": 30,
              }
            }
          >
            Welcome 
          </UNDEFINED>,
          <UNDEFINED
            onPress={[Function]}
            title="Next"
          />,
          <Progress
            amount={0}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeText": [Function],
            "placeholder": "Type here pleass",
            "style": Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "fontSize": 14,
              "padding": 40,
            },
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Welcome ",
            "style": Object {
              "fontSize": 14,
              "padding": 30,
            },
          },
          "ref": null,
          "rendered": "Welcome ",
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "title": "Next",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
